src/util/sss_prctl.h

```diff
@@ -0,0 +1,38 @@
+/*
+    SSSD
+
+    Copyright (C) 2025 Gleb Popov <arrowd@FreeBSD.org>
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef SSS_PRCTL_H_
+#define SSS_PRCTL_H_
+
+#include "config.h"
+
+#if !defined(HAVE_PRCTL) && !defined(HAVE_SYS_PROCCTL_H)
+#define HAVE_NO_PRCTL_WHATSOEVER
```

Comment:
I'd prefer to fail the build if nothing is found. Better at `./configure` step.

Reply:
This would make prctl a requirement. As far as I understand, right now it is an optional feature that enhances security.

Reply:
It's a req since 94352a9fb35e405115733b287972988b02c1a54b - since sssd-2.8 (I didn't realize by that time it was optional req before).

Since nobody complained so far, and this PR adds support for FreeBSD facilities, I think it's better to fix and to spell out the req properly.

Reply:
I agree with Alexey. By the way, since we'll fail at configure step the checks for `HAVE_NO_PRCTL_WHATSOEVER` aren't actually needed.